Bug/nav tabs active state issue

DoRightt opened the pull request.

DoRightt committed:
test/nav-tab - tablink directive tests;

DoRightt committed:
bug/nav-tab - add func types;

DoRightt committed:
bug/nav-tab - change prop types;

DoRightt committed:
bug/nav-tab - fix underline position bug;

DoRightt committed:
bug/nav-tab - navigation fixing;

scDisorder reviewed (changes requested):
```html
<div class="sdk-tab-container__tab"
    [attr.tabIndex]="i"
    [attr.disabled]="tab.disabled || null"
    [class.sdk-tab-container__tab--active]="tab.active"
    [class.sdk-tab-container__tab--fill]="activeTabStyle === 'fill'"
  (click)="selectTab(tab.routerLink, i)">
  <span>{{tab.label}}</span>
  <div class="sdk-tab__background"></div>
</div>
```

to
```html
```html
<a class="sdk-tab-container__tab"
    [tabIndex]="i"
    [disabled]="tab.disabled || null"
    routerLinktActive="tab.active"
    [routerLink]="tab.link"
    [class.sdk-tab-container__tab--fill]="activeTabStyle === 'fill'">
  <span>{{tab.label}}</span>
  <div class="sdk-tab__background"></div>
</a>
```
```

DoRightt committed:
bug/nav-tab - replace divs by link tags;

- replace div tags with click handlers by native link tags with routerLink;
- change animation rule, now it watching at router data #num property to increment & decrement
- change component selectTab method and router events subscription;

DoRightt committed:
bug/nav-tab - change route data property #num -> #index;

DoRightt committed:
bug/nav-tab - add tab opacity on hover;

DoRightt committed:
bug/nav-tab - set few tabs as disabled;
